src/sentry/integrations/github/client.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import absolute_import
+
+import calendar
+import datetime
+import jwt
+import time
+
+from sentry import options
+
+from sentry.integrations.client import ApiClient
+
+
+class GitHubClientMixin(ApiClient):
+    allow_redirects = True
+
+    base_url = 'https://api.github.com'
+
+    def get_last_commits(self, repo, end_sha):
+        # return api request that fetches last ~30 commits
+        # see https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository
+        # using end_sha as parameter
+        return self.get(
+            '/repos/{}/commits'.format(
+                repo,
+            ),
+            params={'sha': end_sha},
+        )
+
+    def compare_commits(self, repo, start_sha, end_sha):
+        # see https://developer.github.com/v3/repos/commits/#compare-two-commits
+        # where start sha is oldest and end is most recent
+        return self.get('/repos/{}/compare/{}...{}'.format(
+            repo,
+            start_sha,
+            end_sha,
+        ))
+
+    def get_pr_commits(self, repo, num):
+        # see https://developer.github.com/v3/pulls/#list-commits-on-a-pull-request
+        # Max: 250 Commits
+        return self.get('/repos/{}/pulls/{}/commits'.format(
+            repo,
+            num
+        ))
+
+
+class GitHubAppsClient(GitHubClientMixin):
+
+    def __init__(self, external_id):
+        self.external_id = external_id
```

Comment:
Is there a reason we don't want the full integration saved and use `integration.external_id` like before? Not sure if we need more than just the `external_id`

Reply:
it might it clearer _what_ external_id we are talking about. I've definitely gotten confused at times for what id is need where

Reply:
this was my suggestion because i don't think we need anything else, and even if we do, i'd rather be more explicit about what we're passing in

Reply:
👍